### PR TITLE
Issue 21 parameter name

### PR DESCRIPTION
**Pull request by Teddy-1000**
Closing #21 
This PR implements the new parameter-name. As a consequence it also changes the KNMI data-loader and multiple integration tests.
During the development of this branch I also discovered a problem if we have multiple time-series with the same parameter-name, but they belong to different time series. The API does not currently separate these, and the order they arrive from gRPC is not consistent, making output unstable in this scenario. This issue is not fixed in this PR.

### CODE REVIEW

File: api/dependencies.py [hunk 1]
```diff
@@ -47,3 +53,36 @@ def get_datetime_range(datetime_string: str | None) -> Tuple[Timestamp, Timestam
...
+async def get_current_parameter_names(ttl_hash=None):
+    """
+    This function get a set of standard_names currently in the datastore
+    The ttl_hash should be a value that is updated at the same frequency
+    we want the lru_cache to be valid for.
+    """
+
+    @lru_cache(maxsize=1)
+    async def async_helper(ttl_hash):
+        del ttl_hash  # make linter think we used this value
```
Comment (Jeffrey-Vervoort-KNMI) [inline api/dependencies.py:67]: If it is only for the linter, I would rather see an inline ignore.

**Comment by Jeffrey-Vervoort-KNMI** [comment]
Some of the tests fail. If those succeed, all the changes look good to me.

File: api/test/test_parse_parameter_name.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "parameter_name_input, expected",
+    [
+        ("atemp:2.0:min:PT10M", HTTPException),
+        ("air_temperature:2.0m:min:PT10M", HTTPException),
+        ("air_temperature:2.0:product:PT10M", HTTPException),
+        ("air_temperature:2.0:min:P10", HTTPException),
+        (":", HTTPException),  # Too few parameters
+        ("::::::", HTTPException),
+    ],
+)  # Too many parameter
```
Comment (Jeffrey-Vervoort-KNMI) [inline api/test/test_parse_parameter_name.py:17]: The colons should be changed to underscores right?
```suggestion
@pytest.mark.parametrize(
    "parameter_name_input, expected",
    [
        ("atemp_2.0_min_PT10M", HTTPException),
        ("air_temperature_2.0m_min_PT10M", HTTPException),
        ("air_temperature_2.0_product_PT10M", HTTPException),
        ("air_temperature_2.0_min_P10", HTTPException),
        ("_", HTTPException),  # Too few parameters
        ("______", HTTPException),
    ],
)  # Too many parameter
```

File: api/dependencies.py (hunk 1 above)
Comment (Teddy-1000) [reply api/dependencies.py]: Add inline pylint ignore

File: api/test/test_parse_parameter_name.py (hunk 2 above)
Comment (Teddy-1000) [reply api/test/test_parse_parameter_name.py]: No, this feature was dropped. Have removed the test.

File: api/requirements.in [hunk 3]
```diff
@@ -12,3 +12,4 @@ covjson-pydantic~=0.2.0
 edr-pydantic~=0.2.0
 shapely~=2.0
 geojson-pydantic~=1.0
+isodate=~0.6
```
[the remark this answers is not in the document]
Comment (Teddy-1000) [reply api/requirements.in]: Also no longer needed as parsing of parameter_name was dropped.

File: datastore/data-loader/client_knmi_station.py [hunk 4]
```diff
@@ -35,15 +36,31 @@ def netcdf_file_to_requests(file_path: Path | str) -> Tuple[List, List]:
...
                 # print(station_id, param_id)
                 param_file = station_slice[param_id]
+                standard_name, level, function, period = (
+                    generate_parameter_name(
+                        (param_file.standard_name if "standard_name" in param_file.attrs else "placeholder"),
+                        "2.0",
+                        param_file.long_name,
+                        param_id,
+                    ),
+                )[0]
```
Comment (Jeffrey-Vervoort-KNMI) [inline datastore/data-loader/client_knmi_station.py:46]:
```suggestion
                standard_name, level, function, period = generate_parameter_name(
                        (param_file.standard_name if "standard_name" in param_file.attrs else "placeholder"),
                        "2.0",
                        param_file.long_name,
                        param_id,
                )
```

File: datastore/data-loader/client_knmi_station.py [hunk 5]
```diff
@@ -61,6 +78,32 @@ def netcdf_file_to_requests(file_path: Path | str) -> Tuple[List, List]:
...
+            period_raw = period_raw[0]
+        else:
+            raise Exception(f"{period_raw}, {long_name}")
+        time, scale = period_raw
+        if scale == "Hours":
+            period = f"PT{time}H"
+        elif scale == "Min":
+            period = f"PT{time}M"
+    else:
+        period = "PT0S"
```
Comment (Jeffrey-Vervoort-KNMI) [inline datastore/data-loader/client_knmi_station.py:102]: Although period is scoped inside the function. I think it is easier to read when placed above the if statement as default for the return statement below.

```suggestion

    period = "PT0S"
    if period_raw := re.findall(r"(\d+) (Hours|Min)", long_name):
        if len(period_raw) == 1:
            period_raw = period_raw[0]
        else:
            raise Exception(f"{period_raw}, {long_name}")
        time, scale = period_raw
        if scale == "Hours":
            period = f"PT{time}H"
        elif scale == "Min":
            period = f"PT{time}M"
```

File: datastore/data-loader/client_knmi_station.py [hunk 6]
```diff
@@ -61,6 +78,32 @@ def netcdf_file_to_requests(file_path: Path | str) -> Tuple[List, List]:
     return observation_request_messages
 
 
+def generate_parameter_name(standard_name, level, long_name, instrument):
```
Comment (Jeffrey-Vervoort-KNMI) [inline datastore/data-loader/client_knmi_station.py:81]: The instrument parameter is not used, could it be removed?

File: datastore/data-loader/client_knmi_station.py (hunk 4 above)
Comment (Teddy-1000) [reply datastore/data-loader/client_knmi_station.py]: The pre-commit did not like this form.

**Review by Jeffrey-Vervoort-KNMI — APPROVED** [review]

**Pull request merged**